bool walk
float64 velocity